### ISSUE TITLE
Questions about ringtones

### ISSUE THREAD

**Issue by August1023** [BelledonneCommunications/linphone-android#1825]
Hi,
What are the functions of "core.setRing core.setRingback core.setRemoteRingbackTone"?
Is "core. setRing" the tone when called?
Is the "core. setRingback" set by the callee so that the main caller can hear the tone?
What is the difference between "core. setRingback" and "core. setRemoteRingbackTone"?
I clearly set the "core. setRing (null)" on the callee, but the main caller can still hear the tone. I am very confused about this.

**Comment by Viish** [comment]
Hi,

[setRing()](https://download.linphone.org/releases/docs/liblinphone/latest/java/org/linphone/core/Core.html#setRing-java.lang.String-) is to define the local ringtone, aka the music you want to hear when you receive a call.

[setRingback()](https://download.linphone.org/releases/docs/liblinphone/latest/java/org/linphone/core/Core.html#setRingback-java.lang.String-) is the music you hear while the device you call is ringing and hasn't been answered yet.

[setRemoteRingbackTone()](https://download.linphone.org/releases/docs/liblinphone/latest/java/org/linphone/core/Core.html#setRemoteRingbackTone-java.lang.String-) is the music you want to stream to the person calling you while the call is ringing until you answer or terminate it, replacing it's own ringback (see above).

**Comment by August1023** [comment]
I see. Thank you very much.

**Comment by MuhammadZohair** [comment]
@Viish I am trying to set the ringtone which is in my res folder, I am not able to do so. when I override the ringtone there is not ring tone at all.

```
core.ring = "android.resource://org.linphone.incomingcall/raw/birdie"
```

I am using .wav format

**Comment by Viish** [comment]
Hi @MuhammadZohair,

Indeed that seems to be an issue from our SDK, I'll see if this can be added for next SDK release.

Cheers,

**Comment by Viish** [comment]
Hi @MuhammadZohair,

FIx has been merged in release/5.3 branch of our SDK, it will be part of 5.3.0 release.

Cheers,

**Comment by MuhammadZohair** [comment]
thankyou @Viish
